File: CS008/P2/P2A5_KELLY_3938535.py
#P2
#John Kelly Jr
#The program will allow the user to test their luck in a game of Rockymon.
#Lecture: Mon/Wed 4:30-5:45
#Lab: Wed 2-2:50

from random import randint
import sys

print('\t\t\t     Rockymon')
print('\t\t\tBy: John Kelly Jr')
name=input('\nPlease enter in your name: ')
print('\nHello,',name,'!')
print('\nRules of the game:')
print('   You will first roll two die. If you roll a 5 or 10 on your first try you win!')
print('   If you roll a 2, 4, or 11 on your first try you lose.')
print('   Otherwise, the number you roll becomes the \"match number.\"')
print('   You must keep rolling until you roll the \"match numer,\" which means you win,')
print('   or a 5, which means you lose.\n')
print('\t\t\t     Good luck!')
#Welcome message

dice = randint(2,12)#random number generator (imported the function up top)

choice=int(input('\nEnter in a "1" if you would like to control the game manually, \n'
            'or enter a "2" for the computer to run the program:'))
      
if choice==1:#user controlled section
    print('\nYou rolled a:', dice)

    if dice == 5 or dice == 10:
        print('Congrats, you win!')
        sys.exit()
    elif dice==2 or dice==4 or dice==11:
        print('I\'m sorry, but you lose.')
        sys.exit()

    print('\nYour new match number is:', dice,'good luck!')


    playGame= input('\nWould you like to play the game again (yes or no)?\n')
    playGame.lower() #make string lower case

    while playGame=='yes':
        dice2 = randint(2,12)
        print('\nThe match number is: ',dice)
        print('You rolled a:', dice2)
        

        if dice2==dice:
            print('Congrats, you win!')
            sys.exit()
        elif dice2==5:
            print('I\'m sorry, but you lose.')
            sys.exit()
        else:
             print('I\'m sorry, but you did not get the match number.')

        playGame= input('\nWould you like to roll again (yes or no)?\n')
        playGame.lower()
            
    
elif choice==2:#computer ran section 
    dice = randint(2,12)    
    w=0
    l=0
    if dice == 5 or dice == 10:
            w+=1
    elif dice==2 or dice==4 or dice==11:
            l+=1

    tries=int(input('\nHow many times would you like to roll the dice?'))    
    for tries in range(0,tries-1):
        dice2 = randint(2,12)

        if dice2==dice:
            w+=1
            print('w')
        elif dice2==5:
            l+=1


    print('You won', w,'times, out of',tries+2,'tries.') 
          

File: CS007/project2/P2A5_KELLY_3938535.java
/* Author: John Kelly Jr
Filename: P2A5_KELLY_3938535
Description: Create a simulations of a craps game.
This particular program will allow the user to try his or her luck. */

import java.util.Scanner; // Import scanner
import java.util.Random; // Import random number generator
import javax.swing.JOptionPane; // Import dialog box

public class P2A5_KELLY_3938535
{
	public static void main(String[] args)
	{

		Scanner input = new Scanner(System.in);
		Random R = new Random();

		// Welcome message

		int tries = JOptionPane.showInputDialog("Please enter in how many"
				+ " times you would like to play:"); // Dialog box


		for(int tried; tried; tried--)





			int dice = R.nextInt(11) + 2; // Two die

			System.out.println("You rolled a: " + dice);


			if(dice == 7 || dice == 11) // A win
			{
				System.out.println("Congratulations you won!");
				System.exit(0);
			}

			else
			{
				if (dice == 2 || dice == 3 || dice == 12) // A loss
				{
					System.out.println("I'm sorry, you lost.");
					System.exit(0);
				}


				else
					System.out.println("Your new point number is: " + dice
						+ ". Good luck!");

			}


			System.out.println("Please enter in \"yes\" if you would like to "
				+ "play again.");
			String confirmation = input.next();

			while (confirmation.equals("yes"))
			{
				int dice2 = R.nextInt(11) + 2; // Two die

				System.out.println("You rolled a: " + dice2);


			if (dice2 == 7)
				{
				System.out.println("I'm sorry, you lost.");
				System.exit(0);
			}

			else if (dice2 == dice)
			{
				System.out.println("Congratulations you won!");
				System.exit(0);
			}

			else
				System.out.println("I'm sorry, you did not win.");

			System.out.println("Please enter in \"yes\" if you would like to "
				+ "play again.");

			String confirmation2 = input.next();

			if (confirmation2.equals(confirmation))

				System.out.println("Best of luck!");

			else
				System.exit(0);

			System.exit(0); // End of dialog box
		}
	}

}
